fvm/evm/handler/invocations.go

```diff
@@ -0,0 +1,34 @@
+package handler
+
+import (
+	"github.com/onflow/cadence/runtime/sema"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func COAOwnershipProofValidator(contractAddress flow.Address, backend types.Backend) func(proof *types.COAOwnershipProofInContext) (bool, error) {
+	return func(proof *types.COAOwnershipProofInContext) (bool, error) {
+		_, err := backend.Invoke(
+			environment.ContractFunctionSpec{
+				AddressFromChain: func(_ flow.Chain) flow.Address {
+					return contractAddress
+				},
+				LocationName: "EVM",
+				FunctionName: "validateCOAOwnershipProof",
+				ArgumentTypes: []sema.Type{
+					types.FlowAddressSemaType,
+					types.PublicPathSemaType,
+					types.SignedDataSemaType,
+					types.KeyIndicesSemaType,
+					types.SignaturesSemaType,
+					types.AddressBytesSemaType,
+				},
+			},
+			proof.ToCadenceValues(),
+		)
+
+		// TODO fix me to only return error
+		return err != nil, nil
```

Comment:
shouldn't this be:
```suggestion
		return err != nil, err
```

Reply:
I'm also addressing this in the follow up PR